src/app/features/Chanels/redux/reducers/index.js

```diff
@@ -0,0 +1,28 @@
+import { GET_CHANNELS, GET_POSTS } from '../actionsTypes';
+
+const initialState = {
+  channels: [],
+  posts: [],
+};
+
+const channelsState = (state = initialState, action) => {
+  console.log(action);
+  switch (action.type) {
+    case GET_CHANNELS: {
+      return {
+        ...state,
+        channels: action.payload,
+      };
+    }
+    case GET_POSTS: {
```

Comment:
I think it may be a good idea to move this action to separate reducer. 

Or store posts per channel id. So that when you open the channel which has been previously open, you see data immediately.